### ISSUE TITLE
Customized accounts' colour doesn't display in the timeline

### ISSUE THREAD

**Issue by SlashTray** [TwidereProject/Twidere-Android#887]
Hi all,

it seems like Twidere doesn't conform to **customized accounts' colour** but rather sticks to predefined account's colour, even if customized later on.

## Expected behavior

I customized the colour of my Mastodon and Twitter accounts. Expecting these colours to display **as customized** to the right-hand side of my timeline.

## Actual behavior

Colours displaying to the right-hand side of my timeline are not the customized ones, but the default/original ones (chosen by Twidere). Restarting Twidere or rebooting my handset doesn't solve the issue.

## Steps to reproduce

1/ Check which colour is assigned by Twidere to a newly defined user account (whichever social network it belongs to I believe).
2/ Customize the colour of this user account by accessing your account's profile > menu > Define Colour option.
3/ Go to your timeline, identify a tweet coming from this user account and check which  colour displays to the right-hand side (currently **not** the customized colour, which is a mistake IMO).

## Extra info

Android version: 6.0

Network type:

- [X] Cellular
- [ ] Wi-Fi

App version: 3.6.21

Build variant:

- [ ] Google
- [X] F-Droid

Micro-blogging service:

- [X] Twitter
- [ ] Fanfou
- [ ] GNU Social / StatusNet
- [X] Mastodon

App settings:

- [?] Custom API
- [?] Custom consumer key/secrets
- [?] Official keys
- [ ] Using [Thumbor](https://github.com/thumbor/thumbor)
- [ ] Connected through proxy
- [X] Media preview enabled

Thank you.

**Comment by mariotaku** [comment]
Color in user profile page is displayed on the left side, while account color on the right side can be changed in Accounts by long-clicking an account.

**Comment by SlashTray** [comment]
Thanks a lot mariotaku, now I can customize my accounts color as desired!
However I'm unable to find that user profile page you mention in the 1st part of your answer. The only trace of a "user" I could find was when I was trying to add new tabs to my timeline/main page ("Likes" for e.g.). Twidere asks me for a user to select, but no user seems to exist... I should be able to find the answer by searching some doc ;)
Still, going to the left pane > Accounts > short-tap an account > top-right menu > Define color, selects a color which means nothing to me... What is this color supposed to be? The "user" one, based on what you said... But what for? Sorry I must be dumb :)
Note : after validating a new "user" color, the previous color which was displaying on the user page doesn't get refreshed automatically.
Apart from these colors mysteries, Twidere is a true jewel, thanks a lot for your hard work!

**Comment by mariotaku** [comment]
User color can be set through tweet page, or user page (that page when you clicked profile image).

> left pane > Accounts > short-tap an account > top-right menu > Define color

This opened user page, and sets user color. In order to set account color, you'll need to long-tap an account and Twidere will bring up a list dialog for you.

**Comment by SlashTray** [comment]
Again, thank you for your time.

It seems I was simply puzzled with the vocabulary you're using. But after manipulating Twidere much more, I think I finally got what you call a "user".
In my understanding, it's an "account" belonging to someone else, right?

If so, calling it a "user" is pretty misleading IMO...
To me, a "user" is a physical person, who owns several "accounts".
"Foreign accounts" (i.e. we're not owning) shouldn't be called "users"...
I think it would be highly valuable to find a better word to describe such accounts.

Back to colors :) Is there a way to set back the default transparent color associated with a "user" (after it was set to another color)?

Thank you.

**Comment by mariotaku** [comment]
@SlashTray Yeah, Twidere calls account belongs to others as 'user'. BTW you can choose 'Clear' when set color for a user to remove the color you've previously set.

**Comment by SlashTray** [comment]
Thank you mariotaku! I should have read more thoroughly...
For other suggestions I have regarding Twidere, I'll post new issues (after checking that they don't yet exist).

Summing up this very issue (#887):
1. Bug (or so): user color doesn't get refreshed automatically right after being changed (cf. user page) (https://github.com/TwidereProject/Twidere-Android/issues/887#issuecomment-313540623).
2. Vocabulary: "user" is too misleading.

Keep up the good work and I'll most probably support you in the near future! You deserve it.

**Comment by bentolor** [comment]
Thanks for asking @SlashTray and thanks  @mariotaku for replying and keeping this open.

I needed to reconfigure/reinstall twidere and this drove me _crazy_ for more than 30 minutes wandering trough settings and changing the different colors.

What it especially made it hard for me is that

* there is a "change color" option if you click a profile, but this has no effect
* there are even more color options which (probably by random) were similar/exactly to the autoselected one

and never would I have been able to figure out the accounts page on my own after being lost for more than 20 minutes....
